activerecord/test/cases/statement_cache_test.rb

```diff
@@ -105,5 +105,18 @@ def test_unprepared_statements_dont_share_a_cache_with_prepared_statements
 
       refute_equal book, other_book
     end
+
+    def test_tablename
+      book = Book.create(name: "my book")
+      assert_equal book, Book.find_by(name: "my book")
+
+      Book.table_name = :birds
+      assert_nil Book.find_by(name: "my book")
+      assert_raise ActiveRecord::RecordNotFound do
+        Book.find(book.id)
+      end
```

Comment:
Ah, testing both forms of cached queries? If so, ensure that the test case communicates its intent, both with the test name and with comments on the assertions it's making.